tests/tarray_ut/tarray_ut.c

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#endif
+
+#include "macro_utils/macro_utils.h"
+
+#include "testrunnerswitcher.h"
+
+void* my_gballoc_malloc(size_t size)
+{
+    return malloc(size);
+}
+
+void my_gballoc_free(void* ptr)
+{
+    free(ptr);
+}
+
+#define ENABLE_MOCKS
+#include "umock_c/umock_c.h"
+#include "c_pal/gballoc_hl.h"
+#include "c_pal/gballoc_hl_redirect.h"
+
+#undef ENABLE_MOCKS
+
+#include "real_gballoc_hl.h"
+
+#include "c_util/tarray.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+#define THANDLE_MALLOC_FUNCTION malloc
+#define THANDLE_FREE_FUNCTION free
+
+/*TARRAY with regular types*/
+TARRAY_TYPE_DECLARE(uint32_t);
+TARRAY_TYPE_DEFINE(uint32_t);
+
+/*TARRAY works with THANDLEs too*/
```

Comment:
Seems that this should be before ln 56. Here there should be a comment that it works with structs

Reply:
Moved to line 56